code/online_challenges/project_euler/problem_036/problem_036.cpp

```diff
@@ -0,0 +1,35 @@
+#include <iostream>
+#include <string>
+
+std::string convertBase2(int number);
+bool isPalindrome(const std::string& str);
+
+int main()
+{
+    int sum = 0;
+    
+    for (int i = 1; i < 1000000; ++i)
+        if (isPalindrome(std::to_string(i)) && isPalindrome(convertBase2(i)))
+            sum += i;
+    std::cout << sum << "\n";
+}
+
+std::string convertBase2(int number)
+{
+    std::string result;
+    
+    while (number > 0)
+    {
+        int currentBit = number % 2;
+        int currentQuotient = number / 2;
+        result += std::to_string(currentBit);
+        number = currentQuotient;
+    }
+    
+    return std::string{ result.rbegin(), result.rend() };
+}
```

Comment:
Why not directly return the result in here?

Just a suggestion below.
There is a clearer version.
```
std::string convertBaseTwo(int number)
{
    std::string res{};

    while (number)
    {
        res += number % 2 ? '1' : '0';
        number >>= 1;
    }

    return res;
}
```

Reply:
I can't directly return result, since the binary would be in reverse, although it may not matter here.

Reply:
Also, this review is now outdated.

Reply:
You are so prudent.